src/runtime-rs/src/shim/src/task_service.rs

```diff
@@ -0,0 +1,848 @@
+                p_m_entry.set_limit(m_data.limit);
+                p_m_entry.set_failcnt(m_data.failcnt);
+                p_m_entry.set_max(m_data.max_usage);
+
+                p_m.set_usage(p_m_entry);
+            }
+            // memory swap_usage
+            if let Some(m_data) = m_stats.swap_usage {
+                let mut p_m_entry = metrics::MemoryEntry::new();
+                p_m_entry.set_usage(m_data.usage);
+                p_m_entry.set_limit(m_data.limit);
+                p_m_entry.set_failcnt(m_data.failcnt);
+                p_m_entry.set_max(m_data.max_usage);
+
+                p_m.set_swap(p_m_entry);
+            }
+            // memory kernel_usage
+            if let Some(m_data) = m_stats.kernel_usage {
+                let mut p_m_entry = metrics::MemoryEntry::new();
+                p_m_entry.set_usage(m_data.usage);
+                p_m_entry.set_limit(m_data.limit);
+                p_m_entry.set_failcnt(m_data.failcnt);
+                p_m_entry.set_max(m_data.max_usage);
+
+                p_m.set_kernel(p_m_entry);
+            }
+
+            for (k, v) in m_stats.stats {
+                match k.as_str() {
+                    "dirty" => p_m.set_dirty(v),
+                    "rss" => p_m.set_rss(v),
+                    "rss_huge" => p_m.set_rss_huge(v),
+                    "mapped_file" => p_m.set_mapped_file(v),
+                    "writeback" => p_m.set_writeback(v),
+                    "pg_pg_in" => p_m.set_pg_pg_in(v),
+                    "pg_pg_out" => p_m.set_pg_pg_out(v),
+                    "pg_fault" => p_m.set_pg_fault(v),
+                    "pg_maj_fault" => p_m.set_pg_maj_fault(v),
+                    "inactive_file" => p_m.set_inactive_file(v),
+                    "inactive_anon" => p_m.set_inactive_anon(v),
+                    "active_file" => p_m.set_active_file(v),
+                    "unevictable" => p_m.set_unevictable(v),
+                    "hierarchical_memory_limit" => p_m.set_hierarchical_memory_limit(v),
+                    "hierarchical_swap_limit" => p_m.set_hierarchical_swap_limit(v),
+                    "total_cache" => p_m.set_total_cache(v),
+                    "total_rss" => p_m.set_total_rss(v),
+                    "total_mapped_file" => p_m.set_total_mapped_file(v),
+                    "total_dirty" => p_m.set_total_dirty(v),
+
+                    "total_pg_pg_in" => p_m.set_total_pg_pg_in(v),
+                    "total_pg_pg_out" => p_m.set_total_pg_pg_out(v),
+                    "total_pg_fault" => p_m.set_total_pg_fault(v),
+                    "total_pg_maj_fault" => p_m.set_total_pg_maj_fault(v),
+                    "total_inactive_file" => p_m.set_total_inactive_file(v),
+                    "total_inactive_anon" => p_m.set_total_inactive_anon(v),
+                    "total_active_file" => p_m.set_total_active_file(v),
+                    "total_unevictable" => p_m.set_total_unevictable(v),
+                    _ => (),
```

Comment:
How about logging this so we know which metrics we're not recording?

Reply:
Fixed in the first revision